src/soc/intel/jasperlake/systemagent.c

```diff
@@ -29,6 +29,9 @@ void soc_add_fixed_mmio_resources(struct device *dev, int *index)
 		{ EPBAR, EP_BASE_ADDRESS, EP_BASE_SIZE, "EPBAR" },
 		{ REGBAR, REG_BASE_ADDRESS, REG_BASE_SIZE, "REGBAR" },
 		{ EDRAMBAR, EDRAM_BASE_ADDRESS, EDRAM_BASE_SIZE, "EDRAMBAR" },
+
+		/* first field (sa_mmio_descriptor.index) is not used, setting to 0: */
+		{ 0, TPM_BASE_ADDRESS, TPM_SIZE, "TPM" },
```

Comment:
I think it should not be done that way. take a look at drivers/pc80/tpm/tis.c:

```c
static void lpc_tpm_read_resources(struct device *dev)
{
	/* Static 5K memory region specified in Kconfig */
	mmio_range(dev, 0, CONFIG_TPM_TIS_BASE_ADDRESS, 0x5000);
}
```

I would check if adding similar thing to drivers/crb/tis.c here:

```c
static struct device_operations __maybe_unused crb_ops = {
	.read_resources = noop_read_resources,
```

Would do the trick.

Reply:
Updated: https://github.com/Dasharo/coreboot/compare/629e34792aed2290429781ee57160132adc1cbbd..26c73446d67c9ec9e6b6d5a48defc168f503b998

Can see the range being reserved on VP1210.